protocol/encoding.go

```diff
@@ -0,0 +1,147 @@
+// Defines methods/functions to encode/decode messages between client and server.
+// Currently this module supports JSON marshal/unmarshal only.
+// Protobuf would be supported in the feature.
+
+package protocol
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+var b64en = base64.StdEncoding.EncodeToString
+
+func MarshalTemporaryBinding(tb *merkletree.TemporaryBinding) ([]byte, error) {
+	if tb == nil {
+		return nil, nil
+	}
+	return json.Marshal(tb)
+}
+
+func MarshalSTR(str *merkletree.SignedTreeRoot) ([]byte, error) {
+	policies, err := json.Marshal(str.Policies)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(&struct {
```

Comment:
Sure! I'm thinking of the messages as a layer of abstraction above the directory, so that a client only needs to "think" in terms of operations they want the server to perform. For example, in the case of a registration, all the client needs to do is send the username, key (and user policies?) to the server, which then handles these inputs and passes them on to the directory. I hope this makes sense. 

Like I said above, I think the request messages already do a god job of representing the operations. But I think the response messages from the server to the client don't define a clear interface yet. I think part of that is that they contain a lot of pieces of data that the client has to parse. Maybe it might be helpful to think of the responses more in terms of client operations as well?